### PR TITLE
Add support for RFC6020 for "if-feature", "when" and "must" statements

### PR DESCRIPTION
**Pull request by gcampbell512**
The "if-feature statement is defined in RFC6020 section 7.18.2 The "when" statement is defined in RFC6020 section 7.19.5 The "must" statement is defined in RFC6020 section 7.5.3

Most of the work is done in apteryx-xml/schema.c.

Added a number of tests to test the new functionality.

### CODE REVIEW

**Comment by gcampbell512** [comment]
This change requires a change to apteryx-xml. It probably will not verify until the apteryx-xml patch is committed. Initially this pull request is so you can see the changes required for apteryx-rest for the apteryx-xml change.

File: rest.c [hunk 1]
```diff
@@ -347,6 +347,8 @@ rest_rpc (int flags, GNode *node, sch_node *schema, json_t *json)
         schflags |= SCH_F_JSON_TYPES;
     if (flags & FLAGS_JSON_FORMAT_NS)
         schflags |= (SCH_F_NS_MODEL_NAME|SCH_F_NS_PREFIX);
+    if (flags & FLAGS_RESTCONF)
```
Comment (carlgsmith) [inline rest.c:350]: if (flags & FLAGS_CONDITIONS)

File: rest.c [hunk 2]
```diff
@@ -644,7 +646,7 @@ rest_api_get (int flags, const char *path, const char *if_none_match, const char
     if (flags & FLAGS_JSON_FORMAT_TYPES)
         schflags |= SCH_F_JSON_TYPES;
     if (flags & FLAGS_RESTCONF)
-        schflags |= SCH_F_IDREF_VALUES;
+        schflags |= SCH_F_IDREF_VALUES | SCH_F_CONDITIONS;
```
Comment (carlgsmith) [inline rest.c:649]: if (flags & FLAGS_CONDITIONS)

File: rest.c [hunk 3]
```diff
@@ -1140,12 +1143,29 @@ rest_api_post (int flags, const char *path, const char *data, int length, const
...
     /* Write the combinded tree to apteryx */
     child->children = tree->children;
+    children = tree->children;
+    tree->children = NULL;
+
+    /* Fixup the parent pointers for the children that now in the root tree */
+    for (GNode *cnode = child->children; cnode; cnode = cnode->next)
+        cnode->parent = child;
+
+    if ((flags & FLAGS_RESTCONF) && (flags & (FLAGS_METHOD_PUT | FLAGS_METHOD_PATCH | FLAGS_METHOD_POST)))
```
Comment (carlgsmith) [inline rest.c:1153]: FLAGS_CONDITIONS

File: rest.c (hunk 1 above)
Comment (gcampbell512) [reply rest.c]: Done

File: rest.c (hunk 2 above)
Comment (gcampbell512) [reply rest.c]: Done

File: rest.c (hunk 3 above)
Comment (gcampbell512) [reply rest.c]: Done

**Review by carlgsmith — APPROVED** [review]

**Pull request merged**